Installation issue:  py-cython

### Steps to reproduce the issue

```console
spack install py-cython %gcc@8.3.0
```

### Information on your system

* **Spack:** 0.15.4-1776-b4ea74c
* **Python:** 3.6.8
* **Platform:** linux-centos7-cascadelake

### Additional information
`spack maintainers py-cython` returns nothing

* [spack-build-out.txt](https://github.com/spack/spack/files/5495167/spack-build-out.txt)

* [spack-build-env.txt](https://github.com/spack/spack/files/5495169/spack-build-env.txt)


### General information

<!-- These boxes can be checked by replacing [ ] with [x] or by clicking them after submitting the issue. -->
- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

So that others don't need to dig into the `spack-build-out.txt`, the specific build failure is:
```
/tmp/root/spack-stage/spack-stage-py-cython-0.29.21-yqkk5btpfl5zub23mqgn26ye2dp7im6q/spack-src/Cython/Plex/Scanners.c:19:10: fatal error: Python.h: No such file or directory
 #include "Python.h"
          ^~~~~~~~~~
```
Can you see if #19768 fixes your problem? If not, my only other guess is that you are using an external package for Python but you don't have `python-devel` installed on your system, so the development headers are missing.

--- a/var/spack/repos/builtin/packages/py-cython/package.py
+++ b/var/spack/repos/builtin/packages/py-cython/package.py
@@ -37,8 +37,8 @@ class PyCython(PythonPackage):
     version('0.23.4',  sha256='fec42fecee35d6cc02887f1eef4e4952c97402ed2800bfe41bbd9ed1a0730d8e')
     version('0.21.2',  sha256='b01af23102143515e6138a4d5e185c2cfa588e0df61c0827de4257bac3393679')
 
-    depends_on('python@2.6:2.8,3.3:', when='@0.23:', type=('build', 'run'))
-    depends_on('python@:2', when='@:0.22', type=('build', 'run'))
+    depends_on('python@2.6:2.8,3.3:', when='@0.23:', type=('build', 'link', 'run'))
+    depends_on('python@:2', when='@:0.22', type=('build', 'link', 'run'))
     depends_on('py-setuptools', type=('build', 'run'))
     depends_on('gdb@7.2:', type='test')
 

